client/failover.go

```diff
@@ -17,10 +17,16 @@ const defaultFailoverGracePeriod = time.Second * 5
 // Note that this client may skip rounds in some cases: e.g. if the group halts
 // for a bit and then catches up quickly, this could jump up to 'current round'
 // and not emit the intermediate values.
+//
+// The passed grace period MUST be <= the group period. Values > the group period
+// will be adjusted to equal the group period.
 func NewFailoverWatcher(core Client, group *key.Group, gracePeriod time.Duration, l log.Logger) Client {
 	if gracePeriod == 0 {
 		gracePeriod = defaultFailoverGracePeriod
 	}
+	if gracePeriod > group.Period {
+		gracePeriod = group.Period
```

Comment:
this seems too long. if i wait to fail over to HTTP client at the end of the period, when i go to the cdn, i'll get the _next_ round and miss the one i've been blocking on completely, right?

Reply:
Yeah, true...what would the alternative be though? We don't know how long the `Get` will take to reach the server. `group.Period - time.Second`? We'd have to also then check `gracePeriod` is not 0 or negative. It's more complexity on an interface that currently doesn't guarantee all randomness anyway. Maybe it would be better to close this PR and open a new one that improves this client to ensure that it _does_ yield all randomness?

Reply:
we decided to just half the group period if 5s was longer than it.